Core/Src/main.c

```diff
@@ -87,27 +99,51 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
+  /* Setting pin 6 as the write pin and setting the pin to high*/
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_6, GPIO_PIN_SET);
 
+  /* Starting the PWM output on the Timer peripheral */
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
-    /* USER CODE END WHILE */
+      /* USER CODE END WHILE */
+
+      /* USER CODE BEGIN 3 */
+
+      // Transmit and receive data using SPI
+      HAL_SPI_TransmitReceive(&hspi1, txData, rxData, 3, HAL_MAX_DELAY);
 
-    /* USER CODE BEGIN 3 */
+      // Extract the ADC value from received data
+      adcVal = ((rxData[1] & 0x03) << 8) | rxData[2];
```

Comment:
How many digits will adcVal carry? What would be the appropriate unit for this variable?